rfcs/0114-nixos-code-of-conduct.md

```diff
@@ -0,0 +1,112 @@
+shepherd-leader:
+related-issues: #98 #102
+---
+
+# Summary
+[summary]: #summary
+
+The NixOS Community needs some foundation of shared values which can be used
+to determine appropriate behavior when contributing and interacting with the community.
+We currently have a [community ethos](https://nixos.org/community/index.html), but
+this only states that we are an inclusive community, but doesn't define
+inappropriate behavior when interacting with the community. This RFC aims
+to establish the shared values, so they can be used to identify behavior
+which is disruptive to the community.
+
+This RFC does not intend to define moderation practices. This is intended to only lay
+a framework of shared values for the community.
+
+# Motivation
+[motivation]: #motivation
+
+The NixOS community lacks a clearly defined way in which someone's behavior may
+be disruptive. A well-defined Code of Conduct gives an explicit
+set of expectations for all contributors, which will also make it easier to identify
+when behavior becomes disruptive.
+
+# Detailed design
+[design]: #detailed-design
+
+These statutes are taken from [rust-lang's Code of Conduct](https://www.rust-lang.org/policies/code-of-conduct),
+which provides a great compromise between explicit behaviors and subjective goals.
+Only minor changes have been made.
+
+- We are committed to providing a friendly, safe and welcoming environment for
+all, regardless of level of experience, gender identity and expression,
+sexual orientation, disability, personal appearance, body size, race, ethnicity, age,
+religion (or lack thereof), socioeconomic status, nationality, or other similar characteristics.
+- Please avoid using overtly sexual aliases or other nicknames that might
+detract from a friendly, safe and welcoming environment for all.
+- Please be kind and courteous. There’s no need to be mean or rude.
+- Respect that people have differences of opinion and that every design or
+implementation choice carries a trade-off and numerous costs. There is seldom a single right answer.
+- Please keep unstructured critique to a minimum. If you have solid ideas
+you want to experiment with, make a fork and see how it works.
+- We will exclude you from interaction if you insult, demean or harass anyone.
+That is not welcome behavior. We interpret the term “harassment” as including the definition in the
+[Citizen Code of Conduct](https://github.com/stumpsyn/policies/blob/b8ee50e7a9fc79c39c936830901f8863d493b7cd/citizen_code_of_conduct.md);
+if you have any lack of clarity about what might be included in that concept,
+please read their definition. In particular, we don’t tolerate behavior that excludes
+people in socially marginalized groups.
+- Private harassment is also unacceptable. No matter who you are, if you feel
+you have been or are being harassed or made uncomfortable by a community member,
+please contact one of the channel ops or any of the NixOS moderation team immediately.
```

Comment:
```suggestion
please contact one of the channel ops or any of the moderation team members immediately.
```

Not sure about what channel ops is supposed to mean. Channel operators? I didn't know we even have such a thing. This text should link to those points of contact. I wouldn't know how to find them.